helix-core/src/main/java/org/apache/helix/api/exceptions/HelixPauseExitsException.java

```diff
@@ -0,0 +1,28 @@
+package org.apache.helix.api.exceptions;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.helix.HelixException;
+
+public class HelixPauseExitsException extends HelixException {
+  public HelixPauseExitsException(String message) {
```

Comment:
Question, why we need an additional exception type here? Could you please share more thoughts?

Reply:
To specifically represents the pause signal has already existed. Could be used for REST api when determining the response code. We can reject the request upon existence or still continue updating the pause znode.

Reply:
I suggest calling it HelixStatusConflictException, or other more general names. Since my suggestion is that we check Maintenance mode too. And I think having 2 or more classes for a similar reason makes sense.
You can always carry the current status in the Exception object for the rest to process.